src/app/stores/actions.ts

```diff
@@ -289,11 +305,17 @@ export function setActiveRouteResponseAction(routeResponseUUID: string) {
  * Add a new route response
  *
  * @param routeReponse - route response to add
+ * @param relativeIndexInList - (optional) indicates where to put the new
+ * route response with relativity to active route response.
  */
-export function addRouteResponseAction(routeReponse: RouteResponse) {
+export function addRouteResponseAction(
+  routeReponse: RouteResponse,
+  relativeIndexInList?: number
```

Comment:
I realize now that you can get rid of this parameters. Unlike a route which can be duplicated when active or not (through the context menu), a route response can only be duplicated when active. 
So the reducer code is enough and there is no need to pass this parameter. Sorry that I made you lose time.

Reply:
I need a parameter to indicate whether this action is a addition or duplication. If it's an addition, new route response is pushed, on the latter it's inserted relatively with the active route response.

I suppose I could not understand your statement with this comment. 

If I were to remove the routeReponse I lose the cloned/freshly created route response.
If I were to remove I lose the duplication indicator for this action.

Are you suggesting to remove both of them?

Reply:
Sorry, you are right, I was probably tired when I wrote this. However, the `relativeIndexInList` is always set to '1' in the `duplicateRouteResponse` method in environment service. So you could either use a boolean instead, and compute the insertion index as you do in the reducer, or compute the index in the service directly.